### PR TITLE
rework how bls messages are kept in a block

### PR DESCRIPTION
**Pull request by whyrusleeping**

### CODE REVIEW

File: data-propagation.md [hunk 1]
```diff
@@ -20,7 +20,8 @@ Blocks are propagated over the libp2p pubsub channel `/fil/blocks`. The followin
 ```sh
 type BlockMessage struct {
   header BlockHeader
-  messages []&SignedMessage
+  secpMsgs []&SignedMessage
```
Comment (dignifiedquire) [inline data-propagation.md]: nit: maybe spell out Messages for consistency

Comment (dignifiedquire) [reply data-propagation.md]: nit: I think secpk is a more common short name

File: data-structures.md [hunk 2]
```diff
@@ -79,6 +79,13 @@ type BlockHeader struct {
 	## worker key to ensure that it is not tampered with after creation
 	blockSig Signature
 } representation tuple
+
+type TxMeta struct {
+  blsMsgs &[&Message]<Sharray>
+
```
Comment (dignifiedquire) [inline data-structures.md]: nit: so much whitespace

File: mining.md [hunk 3]
```diff
@@ -358,17 +366,26 @@ To create a block, the eligible miner must compute a few fields:
...
-    - Note: Messages with BLS signatures should be included as raw `Message` types, and not `SignedMessage`. Their signatures should be gathered up and aggregated for the `BLSAggregate` field.
+  - Separate the messages into BLS signed messages and secp signed messages
+  - For the BLS messages:
+    - Strip the signatures off of the messages, and insert all the bare `Message`s for them into a sharray.
+    - Aggregate all of the bls signatures into a single signature and use this to fill out the `BLSAggregate` field
+  - For the secp messages:
+    - Insert each of the secp `SignedMessage`s into a sharray 
+  - Create a `TxMeta` object and fill each of its fields as follows:
+    - `blsMsgs`: the root cid of the bls messages sharray
+    - `blsAggregate`: the aggregated bls signature
```
Comment (Kubuxu) [inline mining.md]: Refers to `blsAggregate` in `TxMeta`, there isn't `blsAggregate` in `TxMeta`.

**Pull request merged**